src/vsr/grid.zig

```diff
@@ -266,45 +270,91 @@ pub fn GridType(comptime Storage: type) type {
             grid.cache.deinit(allocator);
             grid.blocks_missing.deinit(allocator);
 
-            grid.free_set_encoded.deinit(allocator);
+            grid.free_set_checkpoint.deinit(allocator);
             grid.free_set.deinit(allocator);
 
             grid.* = undefined;
         }
 
-        /// Checkpointing the grid:
-        /// 1. Writes the free set to disk.
-        /// 2. Awaits all pending repair-writes to blocks that were just freed. This guarantees that
-        ///    there are no outstanding writes to (now-)free blocks when we enter the new checkpoint.
-        /// 3. Marks blocks released during the current checkpoint as free and eligible for
-        ///    acquisition in the next checkpoint.
-        /// 4. Marks the free set's own blocks as released (but not yet free).
+        pub fn open(grid: *Grid, callback: *const fn (*Grid) void) void {
+            assert(grid.callback == .none);
+
+            grid.callback = .{ .open = callback };
+            grid.free_set_checkpoint.open(
+                grid,
+                grid.superblock.working.free_set_reference(),
+                free_set_open_callback,
+            );
+        }
+
+        fn free_set_open_callback(free_set_checkpoint: *CheckpointTrailer) void {
```

Comment:
maybe `open_free_set_callback`, so that is lexicographically nested with `open`? I think we have both patterns, and I somewhat consciously chose `open_xxx_callback` naming for free set

Reply:
And we should totally be consitent with `checkpoint_free_set_callback` in the same file